src/main/java/emissary/client/HTTPConnectionFactory.java

```diff
@@ -183,16 +195,50 @@ private static KeyStore buildStore(final String path, final char[] pazz, final S
         if ((path == null) || path.isEmpty()) {
             return null;
         }
+        Path filePath = Paths.get(path);
         final KeyStore keyStore = KeyStore.getInstance(type);
-        try (final InputStream is = Files.newInputStream(Paths.get(path))) {
-            keyStore.load(is, pazz);
+        if (isPemCert(filePath)) {
+            byte[] pemBytes = Files.readAllBytes(filePath);
+            byte[] derBytes = parseDerFromPemCert(pemBytes);
+            X509Certificate x509Certificate = generateCertFromDer(derBytes);
+            keyStore.load(null, pazz);
+            keyStore.setCertificateEntry("alias", x509Certificate);
+        } else {
+            try (final InputStream is = Files.newInputStream(filePath)) {
+                keyStore.load(is, pazz);
+            }
         }
         return keyStore;
     }
 
+    protected static boolean isPemCert(Path filePath) throws IOException {
+        byte[] beginBytes;
+        try (DataInputStream reader = new DataInputStream(Files.newInputStream(filePath))) {
+            int totalNumBytes = reader.available();
+            if (totalNumBytes <= BEGIN_CERT.length() + END_CERT.length()) {
+                return false;
+            }
+            beginBytes = new byte[BEGIN_CERT.length()];
+            reader.read(beginBytes);
+        }
```

Comment:
I need to familiarize myself with `try with resources` more but should it still have a catch? Or just leave it up to the calling function to handle?

Reply:
Is this in reference to converting to readFully when it throws an IOException or in general? I believe that the calling method would just handle the exception in either case. Once the reader is out of scope of the try it closes (so before any catch/finally).